tests/typings/Snapshot.test-d.ts

```diff
@@ -0,0 +1,107 @@
+import type { TypeEqual } from 'ts-expect'
+import { expectType } from 'ts-expect'
+import { INTERNAL_Snapshot as Snapshot } from 'valtio/vanilla'
+
+// Converts object properties to readonly
```

Comment:
@octet-stream do you mind add an explicit test for classes, i.e. add in:

```
class User { firstName = "u1" }
```

And show that it does get recursively `Snapshot`-ified?

Reply:
Sure. I've added a test case for classes.